cmd/antrea-controller/controller.go

```diff
@@ -100,23 +103,50 @@ func run(o *Options) error {
 
 	go apiServer.GenericAPIServer.PrepareRun().Run(stopCh)
 
+	if o.config.EnablePrometheusMetrics {
+		go initializePrometheusMetrics()
+	}
+
 	<-stopCh
 	klog.Info("Stopping Antrea controller")
 	return nil
 }
 
+// Initialize Prometheus metrics collection.
+func initializePrometheusMetrics() {
+
+	nodeName, err := env.GetNodeName()
+	if err != nil {
+		klog.Errorf("Failed to retrieve controller K8S node name, %v", err)
+	}
+
+	klog.Info("Initializing prometheus")
+	gaugeHost := prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "antrea_controller_node",
+		Help: "Antrea controller node name (as a label), typically used in grouping/aggregating stats; " +
+			"The value of the gauge is always set to 1.",
+		ConstLabels: prometheus.Labels{"k8s_nodename": nodeName},
+	})
+	gaugeHost.Set(1)
+	prometheus.MustRegister(gaugeHost)
+}
+
 func createAPIServerConfig(kubeconfig string,
 	bindPort int,
 	addressGroupStore storage.Interface,
 	appliedToGroupStore storage.Interface,
 	networkPolicyStore storage.Interface,
-	controllerQuerier querier.ControllerQuerier) (*apiserver.Config, error) {
+	controllerQuerier querier.ControllerQuerier,
+	authorizeMetrics bool) (*apiserver.Config, error) {
 	// TODO:
 	// 1. Support user-provided certificate.
 	secureServing := genericoptions.NewSecureServingOptions().WithLoopback()
 	authentication := genericoptions.NewDelegatingAuthenticationOptions()
 	authorization := genericoptions.NewDelegatingAuthorizationOptions()
 
+	if authorizeMetrics {
+		authorization.WithAlwaysAllowPaths("/metrics")
```

Comment:
It doesn't really disable metrics when `EnablePrometheusMetrics` is false? It just needs authorization when it's false I think, and other metrics set by apiserver will continue work.
There is a flag `EnableMetrics` in `serverConfig` to disable apiserver's metrics.

Reply:
Understood. I'll set EnableMetrics then.